docbrown/core/src/tgraph_shard.rs

```diff
@@ -179,100 +156,213 @@ impl TGraphShard {
+    pub fn vertices(&self) -> impl Iterator<Item = VertexView> {
         let tgshard = self.rc.clone();
-        let iter: GenBoxed<TVertex> = GenBoxed::new_boxed(|co| async move {
+        let iter: GenBoxed<VertexView> = GenBoxed::new_boxed(|co| async move {
             let g = tgshard.blocking_read();
             let iter = (*g).vertices();
             for vv in iter {
-                let tv: TVertex = vv.into();
-                co.yield_(tv).await;
+                co.yield_(vv).await;
             }
         });
 
-        Box::new(iter.into_iter())
+        iter.into_iter()
     }
 
-    pub fn vertices_window(&self, r: Range<i64>) -> Box<impl Iterator<Item = TVertex> + Send> {
+    pub fn vertices_window(&self, w: Range<i64>) -> impl Iterator<Item = VertexView> {
         let tgshard = self.rc.clone();
-        let iter: GenBoxed<TVertex> = GenBoxed::new_boxed(|co| async move {
+        let iter: GenBoxed<VertexView> = GenBoxed::new_boxed(|co| async move {
             let g = tgshard.blocking_read();
-            let iter = (*g).vertices_window(r);
+            let iter = (*g).vertices_window(w);
             for vv in iter {
-                let tv: TVertex = vv.into();
-                co.yield_(tv).await;
+                co.yield_(vv).await;
             }
         });
 
-        Box::new(iter.into_iter())
+        iter.into_iter()
     }
 
-    pub fn neighbours(&self, v: u64, d: Direction) -> Box<impl Iterator<Item = TEdge> + Send> {
+    pub fn edge(&self, v1: u64, v2: u64) -> Option<EdgeView> {
+        self.read_shard(|tg| tg.edge(v1, v2))
+    }
+
+    pub fn edge_window(&self, v1: u64, v2: u64, w: Range<i64>) -> Option<EdgeView> {
+        self.read_shard(|tg| tg.edge_window(v1, v2, &w))
+    }
+
+    pub fn edges(&self, v: u64, d: Direction) -> impl Iterator<Item = EdgeView> {
         let tgshard = self.rc.clone();
-        let iter: GenBoxed<TEdge> = GenBoxed::new_boxed(|co| async move {
+        let iter: GenBoxed<EdgeView> = GenBoxed::new_boxed(|co| async move {
             let g = tgshard.blocking_read();
-            let iter = (*g).neighbours(v, d);
+            let iter = (*g).edges(v, d);
             for ev in iter {
-                let tv: TEdge = ev.into();
-                co.yield_(tv).await;
+                co.yield_(ev).await;
+            }
+        });
+
+        iter.into_iter()
+    }
+
+    pub fn edges_window(
+        &self,
+        v: u64,
+        w: Range<i64>,
+        d: Direction,
+    ) -> impl Iterator<Item = EdgeView> {
+        let tgshard = self.clone();
+        let iter = gen!({
+            let g = tgshard.rc.blocking_read();
+            let chunks = (*g).edges_window(v, &w, d).map(|e| e.into());
+            let iter = chunks.into_iter();
+            for v_id in iter {
+                yield_!(v_id)
             }
         });
 
-        Box::new(iter.into_iter())
+        iter.into_iter()
+    }
+
+    pub fn edges_window_t(
```

Comment:
At all levels if this function could be renamed to make it a bit clearer what is the different between it and edges_window - as they have such similar method signatures as well.